### PR TITLE
Remove costanalysis preview, it is GA now

### PR DESCRIPTION
**Pull request by matthchr**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [ ] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/aks-preview/azext_aks_preview/_params.py [hunk 1]
```diff
@@ -881,7 +881,6 @@ def load_arguments(self, _):
         c.argument("grafana_resource_id", validator=validate_grafanaresourceid)
         c.argument("enable_windows_recording_rules", action="store_true")
         c.argument("enable_azure_monitor_app_monitoring", is_preview=True, action="store_true")
-        c.argument("enable_cost_analysis", is_preview=True, action="store_true")
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/_params.py:884]: Oh, you'd better keep these parameter declarations (could remove `is_preview=True` though). The declarations in aks-preview are independent from those in official azure-cli.

**Review by FumingZhang — COMMENTED** [review]
Queued [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=99795685&view=results) to validate the change.

- test_aks_create_with_enable_cost_analysis
- test_aks_update_enable_cost_analysis

File: src/aks-preview/azext_aks_preview/_params.py (hunk 1 above)
Comment (matthchr) [reply src/aks-preview/azext_aks_preview/_params.py:884]: Done

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**